spatialos-sdk/src/worker/entity.rs

```diff
@@ -36,36 +45,151 @@ impl Entity {
             user_handle: data_ptr as *mut _,
         };
 
+        self.components.insert(C::ID, raw_data);
+
+        Ok(())
+    }
+
+    pub(crate) fn add_raw(&mut self, component: &Worker_ComponentData) -> Result<(), String> {
```

Comment:
`add_raw` should probably be marked as `unsafe` unless we want to add more validity checks around how `component` is used. Specifically, we don't currently do anything to check that `component.user_handle` is non-null (or otherwise valid) before trying to use it, and I'm not sure it makes a ton of sense to add those checks. The safety of `add_raw` is entirely dependent on the validity of `component`, so it seems incorrect for it to be a "safe" function.

Reply:
Yup, nice catch